### DISCUSSION TITLE
Are tiles in the same tilemap always supposed to have the same z coordinate?

### DISCUSSION THREAD

**Discussion by AntonZelenin** [StarArawn/bevy_ecs_tilemap#420] [General]
Hi, are tiles in the same tilemap always supposed to have the same z coordinate? I assumed all of them have z coordinates the same as the z coordinate of the tilemap itself.
If they could be different, I'd like to know if that's documented somewhere and what z coords I should use for my tilemaps that are on top of each other so that they do not overlap, thanks.

**Comment by rparrett** [comment]
Individual tiles in a tilemap don't have their own z values. The tilemap is rendered together as one big chunk with the z value from the transform in the `TilemapBundle`.

See <https://github.com/StarArawn/bevy_ecs_tilemap/blob/main/examples/layers.rs> for an example of multiple tilemaps layered on top of each other.

You can use any z coordinates you want. Higher values will layer on top of lower values. Keep in mind that by default Bevy's 2d camera shows entities with values between -0.1 and 999.9 or so.

**Reply by StarArawn** [reply]
To add to this isometric tile maps can have z values based on each row of tiles.

**Reply by AntonZelenin** [reply]
There was a bug, but while creating an example for you I figured out it was already fixed, just not released yet. I'll use the main branch as a dependency until there's a new release
